src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/ManagedIndexCoordinator.kt

```diff
@@ -172,13 +179,53 @@ class ManagedIndexCoordinator(
     private fun enable() {
         initBackgroundSweep()
         indexStateManagementEnabled = true
+
+        // Calling initBackgroundSweep() beforehand runs a sweep ensuring that policies removed from indices
+        // and indices being deleted are accounted for prior to re-enabling jobs
+        launch {
+            try {
+                logger.debug("Re-enabling jobs for managed indices")
+                reenableJobs()
+            } catch (e: Exception) {
+                logger.error("Failed to re-enable jobs for managed indices", e)
+            }
+        }
     }
 
     private fun disable() {
         scheduledFullSweep?.cancel()
         indexStateManagementEnabled = false
     }
 
+    private suspend fun reenableJobs() {
+        val clusterStateRequest = ClusterStateRequest()
+            .clear()
+            .metaData(true)
+            .local(false)
+            .indices("*")
+            .indicesOptions(IndicesOptions.strictExpand())
+
+        val response: ClusterStateResponse = client.admin().cluster().suspendUntil { state(clusterStateRequest, it) }
+
+        /*
+         * Iterate through all indices and create update requests to update the ManagedIndexConfig for indices that
+         * meet the following conditions:
+         *   1. Is being managed (has ManagedIndexMetaData)
+         *   2. Does not have a completed Policy
+         *   3. Does not have a failed Policy
+         */
+        val updateManagedIndicesRequests: List<DocWriteRequest<*>> = response.state.metaData.indices.mapNotNull {
+            val managedIndexMetaData = it.value.getManagedIndexMetaData()
+            if (!(managedIndexMetaData == null || managedIndexMetaData.isPolicyCompleted || managedIndexMetaData.isFailed)) {
```

Comment:
Wouldn't this be true for even indices that are not managed at all? We need to check if the index is managed index.

Reply:
This is checking if the ```ManagedIndexMetaData``` exists and that the job is neither completed nor failed. Since we now clear ```ManagedIndexMetaData``` when a policy is removed from an index, the ```ManagedIndexMetaData``` we fetch here should be for managed indices.